.github/workflows/api.yml

```diff
@@ -0,0 +1,48 @@
+# API testing workflow config for CI/CD on Github
+
+name: API continuous integration
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow for main branch on git push
+  push:
+    branches: [master]
+  # Triggers workflow for or any branch on pull requests
+  pull_request:
+    branches:
+      - '**'
+  # Allows running this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow is defined of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "api_test"
+  api_test:
+    strategy:
+      # don't cancel other jobs if one fails
+      fail-fast: false
+      matrix:
+        os: [ubuntu, macos, windows]
```

Comment:
No need to run CI for windows. Windows users are still required to use ubuntu on WSL2.